modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/modal/reassign/bulk/select-assignees-step/SelectAssigneesStepTable.es.js

```diff
@@ -30,27 +30,40 @@ const Item = ({
 
 	const {assigneeId} =
 		reassignedTasks.find(({workflowTaskId}) => workflowTaskId === id) || {};
-
 	const {workflowTaskAssignableUsers: users = []} = data;
+
 	const {assignableUsers = []} =
 		users.find(({workflowTaskId}) => workflowTaskId === id) || {};
 
 	const {name: assigneeName} =
 		assignableUsers.find((assignee) => assignee.id === assigneeId) || {};
 
 	const handleSelect = (newAssignee) => {
-		const filteredTasks = reassignedTasks.filter((task) => task.id !== id);
-
-		if (newAssignee) {
-			filteredTasks.push({
-				assigneeId: newAssignee.id,
-				workflowTaskId: id,
-			});
+		let workflowTasks = reassignedTasks;
+
+		const currentTask = workflowTasks.find(
+			({workflowTaskId}) => workflowTaskId === id
+		);
+
```

Comment:
Hey @naraandrade,

if possible, can you rewrite this logic to reduce the complexity of this code chunk, lines[48-61]? Thx

Reply:
@rafaprax  I'll do it, thanks :)